src/Autofac.Extensions.DependencyInjection/HostBuilderExtensions.cs

```diff
@@ -41,5 +41,27 @@ public static class HostBuilderExtensions
         /// <param name="configurationAction">Action on a <see cref="ContainerBuilder"/> that adds component registrations to the container.</param>
         public static IHostBuilder UseAutofac(this IHostBuilder hostBuilder, Action<ContainerBuilder> configurationAction = null) =>
             hostBuilder.UseServiceProviderFactory(new AutofacServiceProviderFactory(configurationAction));
+
+        /// <summary>
+        /// Register the <see cref="AutofacChildScopeServiceProviderFactory" /> with the Host.
+        /// </summary>
+        /// <param name="hostBuilder">The instance of the <see cref="IHostBuilder"/>.</param>
+        /// <param name="getRootLifetimeScopeFunc">Function to retrieve the root-container instance built using <see cref="ContainerBuilder"/>.</param>
+        /// <param name="configurationAction">Action on a <see cref="ContainerBuilder"/> that adds component registrations to the container.</param>
+        /// <returns>The instance of the <see cref="IHostBuilder"/>.</returns>
+        public static IHostBuilder UseAutofacChildScopeFactory(this IHostBuilder hostBuilder, Func<ILifetimeScope> getRootLifetimeScopeFunc, Action<ContainerBuilder> configurationAction = null) =>
```

Comment:
I think we have enough of these extension methods to warrant creating a separate `Autofac.Extensions.Hosting` package. This will be nicer than having to copy-and-paste the same simple code into every project that uses the hosting abstraction. We can bring this in for now so we have a place for it to move from.

Reply:
Okay. Is there a description somewhere on how to scaffold a new repository? Will you do that? Can I do that? How would that work?

Reply:
I usually just use an existing repository as a template. I can get the first part done as need to integrate with AppVeyor, GitHub teams etc.

Reply:
Okay cool. Then I will create a copy of this project. Just need a github-repository that I can push to then.